backend/main.go

```
@@ -0,0 +1,43 @@
+		apiKey := os.Getenv("GOOGLE_BOOKS_API_KEY")
+		url := fmt.Sprintf("https://www.googleapis.com/books/v1/volumes?q=%s&key=%s", query, apiKey)
+
+		resp, err := http.Get(url)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to fetch data from Google Books API"})
+			return
+		}
+		defer resp.Body.Close()
+
+		var result map[string]interface{}
+		json.NewDecoder(resp.Body).Decode(&result)
```

golangci-lint (1.62.2)

Error return value of `(*encoding/json.Decoder).Decode` is not checked

(errcheck)